awscli/topics/config-vars.rst

```diff
@@ -0,0 +1,209 @@
+CLI config file.
+
+You can specify the following configuration values in the AWS CLI config file:
+
+* ``role_arn`` - The ARN of the role you want to assume.
+* ``source_profile`` - The AWS CLI profile that contains credentials we should
+  use for the initial ``assume-role`` call.
+* ``external_id`` - A unique identifier that is used by third parties to assume
+  a role in their customers' accounts.  This maps to the ``ExternalId``
+  parameter in the ``AssumeRole`` operation.  This is an optional parameter.
+* ``mfa_serial`` - The identification number of the MFA device to use when
+  assuming a role.  This is an optional parameter.  Specify this value if the
+  trust policy of the role being assumed includes a condition that requires MFA
+  authentication. The value is either the serial number for a hardware device
+  (such as GAHT12345678) or an Amazon Resource Name (ARN) for a virtual device
+  (such as arn:aws:iam::123456789012:mfa/user).  
+
+If you do not have MFA authentication required, then you only need to specify a
+``role_arn`` and a ``source_profile``.
+
+When you specify a profile that has IAM role configuration, the AWS CLI
+will make an ``AssumeRole`` call to retrieve temporary credentials.  These
+credentials are then stored (in ``~/.aws/cache``).  Subsequent AWS CLI commands
+will use the cached temporary credentials until they expire, in which case the
+AWS CLI will automatically refresh credentials.
+
+If you specify an ``mfa_serial``, then the first time an ``AssumeRole`` call is
+made, you will be prompted to enter the MFA code.  Subsequent commands will use
+the cached temporary credentials.  However, when the temporary credentials
+expire, you will be re-prompted for another MFA code.
+
+
+Example configuration::
+
+  # In ~/.aws/credentials:
+  [development]
+  aws_access_key_id=foo
+  aws_access_key_id=bar
+
+  # In ~/.aws/config
+  [profile crossaccount]
+  role_arn=arn:aws:iam:...
+  source_profile=development
+
+
+Service Specific Configuration
+==============================
+
+
+aws s3
+------
+
+These values are only applicable for the ``aws s3`` commands.  These
+configuration values are sub values that must be specified under the top level
+``s3`` key.
+
+These are the configuration values you can set for S3:
+
+* ``max_concurrent_requests`` - The maximum number of concurrent requests.
+* ``max_queue_size`` - The maximum number of tasks in the task queue.
+* ``multipart_threshold`` - The size threshold where the CLI uses multipart
+  transfers.
+* ``multipart_chunksize`` - When using multipart transfers, this is the chunk
+  size that will be used.
+
+Example config::
+
+    [profile development]
+    aws_access_key_id=foo
+    aws_secret_access_key=bar
+    s3 =
+      max_concurrent_requests = 20
+      max_queue_size = 10000
+      multipart_threshold = 64MB
+      multipart_chunksize = 16MB
+
+.. I know I'll need to add a proper xref to the s3-config topic.
```

Comment:
One future option might be to have our own role for cross references where in the man page renderer it'll render as the "aws help <topic>" , and in the html docs it will render a link.